src/components/LmsConfigurations/SuccessFactorsConfigForm.jsx

```diff
@@ -0,0 +1,374 @@
+            >
+              <label htmlFor="active">Active</label>
+              <Input
+                type="checkbox"
+                id="active"
+                name="active"
+                className="ml-3"
+                checked={active}
+                onChange={() => this.setState(prevState => ({ active: !prevState.active }))}
+              />
+            </ValidationFormGroup>
+          </div>
+        </div>
+        <div className="row">
+          <div className="col col-4">
+            <ValidationFormGroup
+              for="sapsfBaseUrl"
+              invalid={invalidFields.sapsfBaseUrl}
+              invalidMessage="SAP Success Factors Instance URL is required."
+              helpText="Your SAP Success Factors instance URL. Make sure to include the protocol (ie https/http)"
+            >
+              <label htmlFor="sapsfBaseUrl">SAP Success Factors Instance URL</label>
+              <Input
+                type="text"
+                id="sapsfBaseUrl"
+                name="sapsfBaseUrl"
+                defaultValue={config ? config.sapsfBaseUrl : null}
+              />
+            </ValidationFormGroup>
+          </div>
+        </div>
+        <div className="row">
+          <div className="col col-4">
+            <ValidationFormGroup
+              for="sapsfCompanyId"
+              invalid={invalidFields.sapsfCompanyId}
+              invalidMessage="SAP Success Factors Company Id field is required."
+              helpText="This should match the Company Id as found in SAP Success Factors."
+            >
+              <label htmlFor="sapsfCompanyId">SAP Success Factors Company Id</label>
+              <Input
+                type="text"
+                id="sapsfCompanyId"
+                name="sapsfCompanyId"
+                defaultValue={config ? config.sapsfCompanyId : null}
+              />
+            </ValidationFormGroup>
+          </div>
+        </div>
+        <div className="row">
+          <div className="col col-4">
+            <ValidationFormGroup
+              for="key"
+              invalid={invalidFields.key}
+              invalidMessage="Success Factors' Client Id is required."
+              helpText="Oauth client identifier."
+            >
+              <label htmlFor="key">Client Id</label>
+              <Input
+                type="text"
+                id="key"
+                name="key"
+                defaultValue={config ? config.key : null}
+              />
+            </ValidationFormGroup>
+          </div>
+        </div>
+        <div className="row">
+          <div className="col col-4">
+            <ValidationFormGroup
+              for="secret"
+              invalid={invalidFields.key}
+              invalidMessage="Success Factors' Client Secret is required."
+              helpText="OAuth client secret."
+            >
+              <label htmlFor="secret">Client Secret</label>
+              <Input
+                type="text"
```

Comment:
Should this be "password" instead of text, if it's a secret?

Reply:
TIL that was an input type.
Does that affect anything besides masking the field? Just curious because we don't actually want it encoded/hidden in general.

Reply:
It just masks the field. Basically it only protects against over-the-shoulder and screen capture type attacks. It won't encode the input in anyway, your form should pick it up as a regular form field.

Reply:
Eventually I want a Paragon component that allows the user to show or hide a field like this, but that's a dream for the future.... :)